R/rAccess-utilities.R

```diff
@@ -0,0 +1,127 @@
+
+#' Extract Non-Parent Module Labels to YAML
+#'
+#' @description `r lifecycle::badge("experimental")`
+#' Extracts module labels from a teal modules object, filters out parent modules
+#' (grouping containers), and generates a YAML file with the functional modules.
+#'
+#' @param mods A teal modules object containing the module structure
+#' @param filepath Character string specifying the output YAML file path
+#'
+#' @return Character vector of non-parent module labels
+#'
+#' @examples
+#' \dontrun{
+#' # Extract modules from mods object to YAML file
+#' labels <- extract_modules_to_yaml(mods, "panel_str_modules.yml")
+#' }
+#'
+#' @export
+extract_modules_to_yaml <- function(mods, filepath) {
+  # Recursively extract module labels, excluding parent containers
+  extract_labels <- function(mod_obj) {
+    labels <- character(0)
+    
+    if (inherits(mod_obj, "teal_module")) {
+      # This is a leaf module - extract its label
+      return(mod_obj$label)
+    } else if (inherits(mod_obj, "teal_modules")) {
+      # This is a container - recurse into children
+      for (child in mod_obj$children) {
+        labels <- c(labels, extract_labels(child))
+      }
+    }
+    
+    return(labels)
```

Comment:
```suggestion
    labels
```
R returns the last expression in the function. No need for the explicit return